apps/server/src/modules/files-storage/client/s3-client.adapter.spec.ts

```diff
@@ -85,103 +84,64 @@ describe('S3ClientAdapter', () => {
 				);
 			});
 		});
-
-		describe('WHEN client throws error', () => {
-			const setup = () => {
-				const error = new Error('Bad Request');
-				// @ts-expect-error Testcase
-				client.send.mockRejectedValueOnce(error);
-			};
-
-			it('should throw error from client', async () => {
-				setup();
-
-				await expect(service.createBucket()).rejects.toThrow();
-			});
-		});
 	});
 
 	describe('getFile', () => {
-		describe('WHEN file was received successfully ', () => {
-			const setup = () => {
-				const { pathToFile, config, bytesRange } = createParameter();
-				const resultObj = {
-					Body: { on: () => true },
-					ContentType: 'data.ContentType',
-					ContentLength: 'data.ContentLength',
-					ContentRange: 'data.ContentRange',
-					ETag: 'data.ETag',
-				};
-
-				// @ts-expect-error Testcase
-				client.send.mockResolvedValueOnce(resultObj);
-
-				return { pathToFile, config, bytesRange };
+		const setup = () => {
```

Comment:
In our test structure we define that the when describe must always exists. If the senario description not match well, then you can rewrite the text, or add more then this one.

Reply:
To me the descriptions of the normal flow of the function are always confusing. 'WHEN file was received successfully ' forecasts the result of the test. This is not a scenario description. And there a lot of likewise examples in the code. 

So how would you describe the normal flow?

Reply:
I guess WHEN file was received successfully is misleading and not precise enough. The scenario should relate to the mocked return value of client.send. In this case, it always returns the file successfully because its mocked. The result of the test would be a correct response in the end. As you said, this is unpredictable and depends on the result of the test.

Reply:
I left it now like it is, because I had to roll back most of the changes in the adapter. Now I really just changed the mininum to fulfill the ACs. Maybe some other time, one can have a look at it...